kernel/src/interrupts/legacy_pic.rs

```diff
@@ -0,0 +1,84 @@
+// Thanks to http://www.randomhacks.net/2015/11/16/bare-metal-rust-configure-your-pic-interrupts/
+
+use crate::io::SynchronizedPort;
+use spin::Mutex;
+
+pub static CHAINED_PICS: Mutex<ChainedPics> = Mutex::new(ChainedPics::new((0x20, 0x28)));
+
+/// Used to pause execution temporarily
+pub static IO_WAIT_PORT: SynchronizedPort<u8> = unsafe { SynchronizedPort::new(0x80) };
+
+/// Represents an 8295/8295A PIC (superseded by APIC)
+pub struct Pic {
+    pub offset: u8,
+    pub command_port: SynchronizedPort<u8>,
+    pub data_port: SynchronizedPort<u8>,
+}
+
+impl Pic {
+    const fn new(offset: u8, command_port: SynchronizedPort<u8>, data_port: SynchronizedPort<u8>) -> Self {
+        Pic { offset, command_port, data_port }
+    }
+
+    pub fn initialise(&self) {
+        // Tell the PIC to initialise
+        self.command_port.write(0x10);
+        IO_WAIT_PORT.write(0x0);
+
+        // Set the PICs offset
+        self.write(self.offset);
+    }
+
+    #[inline]
+    pub fn write(&self, value: u8) {
+        self.data_port.write(value);
+        IO_WAIT_PORT.write(0x0);
+    }
+}
+
+/// Represents two [Pic]s chained together as they are in the hardware
+pub struct ChainedPics {
+    inner: [Pic; 2],
+}
+
+impl ChainedPics {
+    pub const fn new(offsets: (u8, u8)) -> Self {
+        unsafe {
+            ChainedPics {
+                inner: [
+                    Pic::new(offsets.0,
+                             SynchronizedPort::new(0x20),
```

Comment:
Formatting should probably be consistent with below

Reply:
The inclusion of this file in actually an accident.